velox/exec/tests/HashJoinTest.cpp

```diff
@@ -638,6 +638,20 @@ class HashJoinBuilder {
         memory::spillMemoryPool()->stats().peakBytes;
     TestScopedSpillInjection scopedSpillInjection(spillPct);
     auto task = builder.assertResults(referenceQuery_);
+    // Wait up to 5 seconds for all the task background activities to complete.
+    // Then we can collect the stats from all the operators.
+    uint64_t totalTaskWaitTimeUs{0};
+    while (task.use_count() != 1) {
```

Comment:
Can we use something like Task::taskCompletionFuture(uint64_t maxWaitMicros) ?

Reply:
It won't work as task terminate set the state to finish first and then close the drivers, and there will be a time race there.